internal/tools/uvmboot/wcow.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/hcsoci"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+var (
+	wcowDockerImage string
+	wcowCommandLine string
+)
+
+var wcowCommand = cli.Command{
+	Name:  "wcow",
+	Usage: "boot a WCOW UVM",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:        "exec",
+			Usage:       "Command to execute in the UVM.",
+			Destination: &wcowCommandLine,
+		},
+		cli.StringFlag{
+			Name:        "docker-image",
+			Usage:       "Docker image to use for the UVM image",
+			Destination: &wcowDockerImage,
+		},
+	},
+	Action: func(c *cli.Context) error {
+		runMany(c, func(id string) error {
+			options := uvm.NewDefaultOptionsWCOW(id, "")
+			setGlobalOptions(c, options.Options)
+			if wcowDockerImage == "" {
+				wcowDockerImage = "mcr.microsoft.com/windows/nanoserver:1809"
+			}
+			layers, err := getLayers(wcowDockerImage)
+			if err != nil {
+				return err
+			}
+			tempDir, err := ioutil.TempDir("", "uvmboot")
+			if err != nil {
+				return err
+			}
+			defer os.RemoveAll(tempDir)
+			options.LayerFolders = append(layers, tempDir)
+			vm, err := uvm.CreateWCOW(options)
+			if err != nil {
+				return err
+			}
+			defer vm.Close()
+			if err := vm.Start(); err != nil {
+				return err
+			}
+			if wcowCommandLine != "" {
```

Comment:
Did you mean to terminate if `wcowCommandLine == ""`? This will wait forever

Reply:
Yeah probably it should terminate.